src/groupByKey.ts

```diff
@@ -2,6 +2,15 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  return items.reduce((result, item) => {
+    const groupKey = item[key] as string;
+
+    if (!result[groupKey]) {
+      result[groupKey] = [];
+    }
+    result[groupKey].push(item);
+
+    return result;
+  }, {} as GroupsMap<T>);
```

Comment:
When providing an initial value for the `reduce` function, it's better to explicitly state the type for readability and to avoid any potential type issues. You could define the initial object outside of the `reduce` call with the `GroupsMap<T>` type.